sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java

```diff
@@ -179,21 +247,25 @@ public TableClient createTable(String tableName) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<TableClient> createTableWithResponse(String tableName, Duration timeout, Context context) {
-        return blockWithOptionalTimeout(createTableWithResponse(tableName, context), timeout);
+        final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
Creating and shutting down a new Scheduled Executor service will likely add good chunk of overhead.
We should create a static scheduler which shuts down gracefully.